[Feature]: Add support for DevLogin (#192)
neoforged/NeoGradle: README.md

```properties
neogradle.subsystems.tools.jst=<artifact coordinate for jst cli tool>
```
### DevLogin

> This tool is used by the runs subsystem to enable logged in plays on all client runs.

Reviewer: ```suggestion
This tool is used by the runs subsystem to enable Minecraft authentification in client runs.
```

The following properties can be used to configure the DevLogin tool:
```properties
neogradle.subsystems.tools.devLogin=<artifact coordinate for devlogin cli tool>
```
More information on the relevant tool, its released version and documentation can be found here: [DevLogin by Covers1624](https://github.com/covers1624/DevLogin)

## DevLogin
The DevLogin tool is a tool that allows you to log in to a Minecraft account without having to use the Minecraft launcher, during development.
This tool is used by the runs subsystem to enable logged in plays on all client runs.
The tool can be configured using the following properties:
```properties
neogradle.subsystems.devLogin.enabled=<true/false>
```

> By default, the subsystem is enabled, and it will ask you to log in when you run a client run.

Reviewer: This is contradicting with the lines below. Is it, or is it not enabled?

> If you want to disable this, you can set the property to false, and then you will not be asked to log in, but use a random none signed in dev account.

Reviewer: No comma after if clauses. Also, `non`, not `none`.

Additionally, you can configure whether to add the relevant repositories to your buildscript:
```properties
neogradle.subsystems.devLogin.addRepositories=<true/false>
``` 
By default, the repositories are added, but if you want to disable this, you can set the property to false.
By doing so you are responsible for adding a maven repository that provides the dev login CLI artifact.

### Per run configuration
If you want to configure the dev login tool per run, you can do so by setting the following properties in your run configuration:
```groovy
runs {
    someRun {

>         shouldUseDevLogin true

Reviewer 2: You forgot to update this line to match the new property `useDevLogin`.

    }
}
```
This will enable the dev login tool for this run. 
By default, the dev login tool is disabled, and can only be enabled for client runs.

> [!WARNING]
> If you enable the dev login tool for a none client run, you will get an error message.

## Centralized Cache
NeoGradle has a centralized cache that can be used to store the decompiled Minecraft sources, the recompiled Minecraft sources, and other task outputs of complex tasks.
The cache is enabled by default, and can be disabled by setting the following property in your gradle.properties:
```properties
net.neoforged.gradle.caching.enabled=false
```

You can clean the artifacts that are stored in the cache by running the following command:
```shell
./gradlew cleanCache
```

> This command is also automatically ran, when you run the clean task.

Reviewer: This sentence is in the passive voice, which requires a past participle verb (`run` being the past participle of `run`)

> The command will check if the stored artifact count is higher than the configured threshold, and if so remove the oldest artifacts until the count is below the threshold.

Reviewer: ```suggestion
The command will check if the stored artifact count is higher than the configured threshold, and if so, remove the oldest artifacts until the count is below the threshold.
```

```properties
neogradle.subsystems.tools.jst=<artifact coordinate for jst cli tool>
```
### DevLogin
This tool is used by the runs subsystem to enable Minecraft authentication in client runs.
The following properties can be used to configure the DevLogin tool:
```properties
neogradle.subsystems.tools.devLogin=<artifact coordinate for devlogin cli tool>
```
More information on the relevant tool, its released version and documentation can be found here: [DevLogin by Covers1624](https://github.com/covers1624/DevLogin)

## DevLogin

> The DevLogin tool is a tool that allows you to log in to a Minecraft account without having to use the Minecraft launcher, during development.

Reviewer 2: As I understand it, DevLogin works when the developer is logged into the vanilla Minecraft launcher, correct? That should be clarified here.

```properties
neogradle.subsystems.tools.jst=<artifact coordinate for jst cli tool>
```
### DevLogin
This tool is used by the runs subsystem to enable Minecraft authentication in client runs.
The following properties can be used to configure the DevLogin tool:
```properties
neogradle.subsystems.tools.devLogin=<artifact coordinate for devlogin cli tool>
```
More information on the relevant tool, its released version and documentation can be found here: [DevLogin by Covers1624](https://github.com/covers1624/DevLogin)

## DevLogin
The DevLogin tool is a tool that allows you to log in to a Minecraft account without having to use the Minecraft launcher, during development.

> By default, it will use the credentials stored by the vanilla launcher, however if you are not logged in there, it will offer you the ability to log in directly from the console.

Reviewer 3: DevLogin does not have the ability to use the vanilla launcher credentials. It purely implements the device auth flow. If someone can figure out how they store the creds, we can absolutely add support for it.

```properties
neogradle.subsystems.tools.jst=<artifact coordinate for jst cli tool>
```
### DevLogin
This tool is used by the runs subsystem to enable logged in plays on all client runs.
The following properties can be used to configure the DevLogin tool:
```properties

> neogradle.subsystems.tools.devLogin=<artifact coordinate for devlogin cli tool>

Reviewer 2: Perhaps this should document the GitHub repository of the DevLogin tool.

```

## DevLogin
The DevLogin tool is a tool that allows you to log in to a Minecraft account without having to use the Minecraft launcher, during development.
This tool is used by the runs subsystem to enable logged in plays on all client runs.
The tool can be configured using the following properties:
```properties
neogradle.subsystems.devLogin.enabled=<true/false>
```
By default, the subsystem is enabled, and it will ask you to log in when you run a client run.
If you want to disable this, you can set the property to false, and then you will not be asked to log in, but use a random none signed in dev account.

Additionally, you can configure whether to add the relevant repositories to your buildscript:
```properties
neogradle.subsystems.devLogin.addRepositories=<true/false>
``` 
By default, the repositories are added, but if you want to disable this, you can set the property to false.
By doing so you are responsible for adding a maven repository that provides the dev login CLI artifact.

### Per run configuration
If you want to configure the dev login tool per run, you can do so by setting the following properties in your run configuration:
```groovy
runs {
    someRun {

>         shouldUseDevLogin true

Reviewer 2: Why is it called `shouldUseDevLogin`? Why not `useDevLogin`?

    }
}
```
This will enable the dev login tool for this run. 
By default, the dev login tool is disabled, and can only be enabled for client runs.

> [!WARNING]

> > If you enable the dev login tool for a none client run, you will get an error message.

Reviewer: You need the `non` prefix, not the `none` pronoun.
